core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -85,7 +89,22 @@ public AllocationService(Settings settings, AllocationDeciders allocationDecider
         if (withReroute) {
             reroute(allocation);
         }
-        return new RoutingAllocation.Result(true, new RoutingTable.Builder().updateNodes(routingNodes).build().validateRaiseException(clusterState.metaData()));
+        RoutingTable routingTable = new RoutingTable.Builder().updateNodes(routingNodes).build().validateRaiseException(clusterState.metaData());
+        RoutingAllocation.Result result = new RoutingAllocation.Result(true, routingTable);
+
+        final int maxNumberOfLoggedShards = 10;
+        String startedShardsAsString = startedShards
+                .stream()
+                .limit(maxNumberOfLoggedShards)
+                .map(s -> String.format(Locale.ROOT, "[index: [%s], shard id: [%d]]", s.getIndex(), s.getId()))
+                .collect(Collectors.joining(", "));
+
```

Comment:
I think the Java 8 stream API makes this code quite readable. I am aware that the backport to 2.x needs to be in Java 7 style.